### PR TITLE
Update EIP-1: Permit links to Execution Specs

### PR DESCRIPTION
**Pull request by pcaversaccio**
This PR adds the [Ethereum Execution Client Specifications](https://github.com/ethereum/execution-specs) as well as the [Ethereum Yellow Paper](https://github.com/ethereum/yellowpaper) as permitted links to EIP-1. The related draft EIP triggering this PR is https://github.com/ethereum/EIPs/pull/6269.

### CODE REVIEW

File: EIPS/eip-1.md [hunk 1]
```diff
@@ -194,6 +194,24 @@ Other than the specific exceptions listed below, links to external resources **S
...
+```
+
+Which renders to:
+
+[Ethereum Execution Client Specifications](https://github.com/ethereum/execution-specs/blob/9a1f22311f517401fed6c939a159b55600c454af/README.md)
+
+Permitted Execution Client Specifications URLs must anchor to a specific commit, and so must match this regular expression:
+
+```regex
+^https://github.com/ethereum/execution-specs/blob/[0-9a-f]{40}/.*$
```
Comment (Pandapip1) [inline EIPS/eip-1.md:212]:
```suggestion
Permitted Execution Client Specifications URLs must anchor to a specific commit, and so must match this regular expression:

```regex
^(https://github.com/ethereum/execution-specs/blob/[0-9a-f]{40}/.*|https://github.com/ethereum/execution-specs/tree/[0-9a-f]{40}/.*)$
```
This allows the file tree view (which is useful in your case).

File: EIPS/eip-1.md [hunk 2]
```diff
@@ -230,6 +248,24 @@ Permitted Networking Specifications URLs must anchor to a specific commit, and s
...
+```
+
+Which renders to:
+
+[Ethereum Yellow Paper](https://github.com/ethereum/yellowpaper/blob/1016c0603062b76388e3c3c19786cd5f9ca9ac61/paper.pdf)
+
+Permitted Ethereum Yellow Paper URLs must anchor to a specific commit, and so must match this regular expression:
+
+```regex
+^https://github.com/ethereum/yellowpaper/blob/[0-9a-f]{40}/.*$
```
Comment (Pandapip1) [inline EIPS/eip-1.md:266]:
```suggestion
^(https://github.com/ethereum/yellowpaper/blob/[0-9a-f]{40}/.*|https://github.com/ethereum/yellowpaper/tree/[0-9a-f]{40}/.*)$
```
Same here.

File: EIPS/eip-1.md (hunk 1 above)
Comment (pcaversaccio) [reply EIPS/eip-1.md:212]: oh yes, that's awesome - didn't know whether it was allowed to change the regex like that. Do you want me to update the example link as well (even though it's still a valid example)?

**Review by Pandapip1 — APPROVED** [review]
+1 from me

(shouldn't automerge, if it does, it should be rolled back)

**Comment by lightclient** [comment]
I am +1 on execution spec and -1 on yellow paper.

**Comment by pcaversaccio** [comment]
> I am +1 on execution spec and -1 on yellow paper.

I need the Yellow Paper as a reference in order to make https://github.com/ethereum/EIPs/pull/6269 backwards-compatible.

**Comment by pcaversaccio** [comment]
> There are going to be some changes to the EIP process that will affect how it interacts with the execution-specs.

Sorry I'm confused - 2 weeks ago you ask me [here](https://github.com/ethereum/EIPs/pull/6269#issuecomment-1377434872) to open a PR to EIP-1 to add the proposed links. Now it seems the world has completely changed since then. I would appreciate some transparency about what is happening on the EIP process side since it significantly impacts my own EIP proposal. I know you state that you don't know what changes will happen, but could you explain quickly at least what process you're running? That would be appreciated.

**Comment by SamWilsn** [comment]
> Sorry I'm confused - 2 weeks ago you ask me [here](https://github.com/ethereum/EIPs/pull/6269#issuecomment-1377434872) to open a PR to EIP-1 to add the proposed links.

Yeah, sorry about the confusion. It's easier to talk about governance issues when we have a concrete pull request to discuss.

---

After discussing on today's EIPIP meeting, I think we're generally positive on adding this. I'll bug @lightclient in case he _absolutely_ doesn't want to allow Yellow Paper links.

**Comment by pcaversaccio** [comment]
Thanks for the heads-up - a quick question for discussion here. As you might know, I also plan to draft an EIP on _Full Ethereum Equivalence_ and for that, I would have to partially use the [`Ethereum Consensus Tests`](https://github.com/ethereum/tests). Now I wanted to ask whether there is a clear objection from your side to adding this as well. As background, many zkEVMs and optimistic Rollups use this metric to showcase compatibility or even equivalence.

**Comment by Pandapip1** [comment]
> Now I wanted to ask whether there is a clear objection from your side to adding this as well.

Please add that in a *separate* PR. Adding it to this PR will make it harder to reach a consensus (or a compromise, if need be).

**Comment by pcaversaccio** [comment]
> > Now I wanted to ask whether there is a clear objection from your side to adding this as well.
> 
> Please add that in a _separate_ PR. Adding it to this PR will make it harder to reach a consensus (or a compromise, if need be).

Sure, will do it once I drafted the EIP where this is needed.

**Comment by SamWilsn** [comment]
I suppose I have no strong objection to linking to the tests repository.

**Comment by pcaversaccio** [comment]
@SamWilsn but would like to have it as part of this PR already? As pointed out by @Pandapip1, I don't want to make it harder to find a consensus for this PR :)

**Comment by Pandapip1** [comment]
@pcaversaccio I'm a bit concerned that the more links that are allowed in a single PR, the harder it will be to get any of them merged. Already, the Yellow Paper is the only thing that's blocking this PR.

**Comment by pcaversaccio** [comment]
@Pandapip1 agreed - so I will refrain from submitting another link.

**Comment by Pandapip1** [comment]
In the meantime, feel free to submit more PRs!

**Comment by pcaversaccio** [comment]
I will submit the PR with the additional links once I drafted the EIP for _Full Ethereum Equivalence_. However, before doing this we need first to find an agreement in https://github.com/ethereum/EIPs/pull/6269.

**Comment by wschwab** [comment]
I honestly think linking to the Yellow Paper should also be allowed, fwiw. Even if the Execution Specs are the canonical way moving forward, the Yellow Paper was the original specification, and was still at the least a definitive part of defining Ethereum until the migration to the Specs began (all of the problems with it notwithstanding).

imho it's easy to envision EIPs that make reference to the Yellow Paper to validate assertions or define what expected behavior is within the context of an EIP. As such, the logical course would be to allow links to the Paper, again imho.

**Comment by pcaversaccio** [comment]
@lightclient @SamWilsn @Pandapip1 - any update on how we progress from here now?

**Comment by Pandapip1** [comment]
We're just waiting on more editors to approve or voice their objections.

**Comment by xinbenlv** [comment]
I am in strong favor of allowing both Yellow Paper and Execution Spec. If no full consensus, I am ok to merge a PR to add Exec-Specs first and wait for consensus to merge another PR to add Yellow Paper.

**Comment by lightclient** [comment]
I absolutely do not think the yellow paper should be an acceptable resource in the EIPs repository.

**Review by SamWilsn — APPROVED** [review]

File: EIPS/eip-1.md [hunk 3]
```diff
@@ -230,6 +248,24 @@ Permitted Networking Specifications URLs must anchor to a specific commit, and s
...
+Which renders to:
+
+[Ethereum Yellow Paper](https://github.com/ethereum/yellowpaper/blob/1016c0603062b76388e3c3c19786cd5f9ca9ac61/paper.pdf)
+
+Permitted Ethereum Yellow Paper URLs must anchor to a specific commit, and so must match this regular expression:
+
+```regex
+^(https://github.com/ethereum/yellowpaper/blob/[0-9a-f]{40}/.*|https://github.com/ethereum/yellowpaper/tree/[0-9a-f]{40}/.*)$
+```
+
```
[the remark this answers is not in the document]
Comment (pcaversaccio) [reply EIPS/eip-1.md]: This decision will break my definition in https://github.com/ethereum/EIPs/pull/6269 wrt backward-compatibility...

**Review by lightclient — APPROVED** [review]

**Pull request merged**